### PR TITLE
displayer for Illuminate/Laravel Custom HTTP Error Pages

### PR DESCRIPTION
**Pull request by younes0**

### CODE REVIEW

File: src/Displayers/IlluminateHtmlDisplayer.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+     * @param \Exception $exception
+     * @param string     $id
+     * @param int        $code
+     * @param string[]   $headers
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function display(Exception $exception, $id, $code, array $headers)
+    {
+        if (view()->exists("errors.{$code}")) {
```
Comment (GrahamCampbell) [inline src/Displayers/IlluminateHtmlDisplayer.php:45]: we can't use global functions like this in a package, in all cases, so i'll sort that out

File: src/Displayers/IlluminateHtmlDisplayer.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            return new Response(view()->make("errors.{$code}"), $code, array_merge($headers, [
```
Comment (GrahamCampbell) [inline src/Displayers/IlluminateHtmlDisplayer.php:46]: same

File: src/Displayers/IlluminateHtmlDisplayer.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                'Content-Type' => $this->contentType(),
+            ]));
+
+        } else {
+            return (new SymfonyDisplayer(config('app.debug')))->createResponse($exception);
```
Comment (GrahamCampbell) [inline src/Displayers/IlluminateHtmlDisplayer.php:51]: same

Comment (GrahamCampbell) [inline src/Displayers/IlluminateHtmlDisplayer.php:51]: actually, this whole line is wrong

Comment (GrahamCampbell) [inline src/Displayers/IlluminateHtmlDisplayer.php:51]: a "displayer" can either be for debug mode, or not

Comment (GrahamCampbell) [inline src/Displayers/IlluminateHtmlDisplayer.php:51]: `view()->exists("errors.{$code}")` needs moving to the "can display" method

**Comment by GrahamCampbell** [comment]
I think I'll actually do this a different way, looking at the implementation in more detail.

**Comment by younes0** [comment]
Okay no worries, willing to learn.
May I ask you why we can't use global functions in packages? The code was actually taken from `Illuminate\Foundation\Exceptions\Handler`

**Comment by GrahamCampbell** [comment]
Just out of principle. This package needs to work everywhere, including on non-laravel installs. As far as I'm concerned I'd like all my laravel packages to work everywhere.

**Comment by GrahamCampbell** [comment]
So provided people have implementations of the needed contracts, they can use my packages in any context they want.

**Comment by younes0** [comment]
Okay I see, thanks

**Comment by GrahamCampbell** [comment]
I'll have to bump the major version for this since I'll need to pass the code to canDisplay.

**Comment by GrahamCampbell** [comment]
I'll send a PR with the changes so we can both check them in a few mins.

**Comment by GrahamCampbell** [comment]
How does https://github.com/GrahamCampbell/Laravel-Exceptions/pull/29 look?

**Pull request closed without merging**